invokeai/app/services/config.py

```diff
@@ -0,0 +1,364 @@
+# Copyright (c) 2023 Lincoln Stein (https://github.com/lstein)
+
+'''Invokeai configuration system.
+
+Arguments and fields are taken from the pydantic definition of the
+model.  Defaults can be set by creating a yaml configuration file that
+has top-level keys corresponding to an invocation name, a command, or
+"globals" for global values such as `xformers_enabled`. Currently
+graphs cannot be configured this way, but their constituents can be.
+
+[file: invokeai.yaml]
+
+ globals:
+   nsfw_checker: False
+   max_loaded_models: 5
+ 
+ txt2img:
+   steps: 20
+   scheduler: k_heun
+   width: 768
+ 
+ img2img:
+   width: 1024
+   height: 1024
+
+The default name of the configuration file is `invokeai.yaml`, located
+in INVOKEAI_ROOT. You can use any OmegaConf dictionary by passing it
+to the config object at initialization time:
+
+ omegaconf = OmegaConf.load('/tmp/init.yaml')
+ conf = InvokeAIAppConfig(conf=omegaconf)
+
+By default, InvokeAIAppConfig will parse the contents of argv at
+initialization time. You may pass a list of strings in the optional
+`argv` argument to use instead of the system argv:
+
+ conf = InvokeAIAppConfig(arg=['--xformers_enabled'])
+
+It is also possible to set a value at initialization time. This value
+has highest priority.
+
+ conf = InvokeAIAppConfig(xformers_enabled=True)
+
+Any setting can be overwritten by setting an environment variable of
+form: "INVOKEAI_<command>_<value>", as in:
+
+  export INVOKEAI_txt2img_steps=30
+
+Order of precedence (from highest):
+   1) initialization options
+   2) command line options
+   3) environment variable options
+   4) config file options
+   5) pydantic defaults
+
+Typical usage:
+
+ from invokeai.app.services.config import InvokeAIAppConfig
+ from invokeai.invocations.generate import TextToImageInvocation
```

Comment:
I tried out this typcial usage example as a script. To get working, I needed to change 
`invokeai.invocations.generate` to `invokeai.app.invocations.generate`
And `TextToImageInvocation()` to `TextToImageInvocation(id="dummyid")`

Reply:
> Trying out web UI: invokeai.py --web ==> works well (though only tested TextToImage) invokeai-new.py --web ==> server crashed on startup with error:
> 
> ` File "...../invokeai/app/services/graph.py", line 1179, in validate_exposed_nodes graph = values['graph'] KeyError: 'graph' ERROR: Application startup failed. Exiting.`
> 
> It's possible there's something weird in my setup that's triggering this only for me, but I don't know what that would be.
> 
> I also ran pytest for the three test files modified in this PR, and they passed all tests.
> 
> On Ubuntu 22.04, Nvidia A5000 card

Is this crash specific to the PR, or present in `main` as well? I launch the new web server using the command `invokeai-node-web` and it seems to be working in the PR and main.

Reply:
It's specific to the PR, but... <br>
So it looks like the problem was partly in how I started up the web server. I've been using `python scripts/invoke-new.py --web` instead of `invoke-node-web` (which I didn't know about). Just tried `invoke-node-web` and it works on both PR and main. Weirdly, I then switched back to trying PR with `python scripts/invoke-new.py --web` and now it works. Maybe using `invoke-node-web` once bootstrapped something? Anyway seems like possible funky setup on my end combined with using a different entry point. Possibly scripts/invoke-new.py should be deprecated or updated? But in any case I'll go ahead and approve the PR.